apps/evans/components/core/AudioPlayer.tsx

```diff
@@ -171,3 +174,10 @@ const AudioPlayer: React.FC<Props> = ({ tracks }) => {
 };
 
 export default AudioPlayer;
+
+function duration(element: HTMLAudioElement | null, serverDuration: number): number {
+  if (element && !Number.isNaN(element.duration)) {
```

Comment:
this seemed to fix the occasional `NaN`s we were still seeing, sometimes we have a valid audio element, but the `.duration` prop can be NaN, which is documented. but we prefer the actual audio element duration over the server-supplied number, because i noticed scrubbing to the end of very long audios, sometimes the server/duration and the element duration were off by a second or so